provider/cmd/pulumi-resource-command/schema.json

```diff
@@ -497,6 +497,26 @@
                     "description": "The password to use in case the private key is encrypted.",
                     "type": "string"
                 },
+                "proxyHost": {
+                    "description": "The address of the bastion host to connect to.",
+                    "type": "string"
+                },
+                "proxyPassword": {
+                    "description": "The password we should use for the bastion host connection.",
+                    "type": "string"
+                },
+                "proxyPort": {
+                    "default": 22,
+                    "type": "integer"
+                },
+                "proxyPrivateKey": {
+                    "description": "The contents of an SSH key to use for the bastion host to setup the connection. This takes preference over the password if provided.",
+                    "type": "string"
+                },
+                "proxyUser": {
+                    "description": "The user that we should use for the bastion host connection.",
+                    "type": "string"
+                },
```

Comment:
I think this is probably good the way it is - but a few related questions/thoughts:
1. Should this be nested under a single `proxy` property?
2. There is a lot of overlap with `Connection` itself - should this even be a nested Connection object?
3. There are a couple things on `Connection` that aren't here - `privateKeyPassword`, `dialErrorLimit`.  Are either of these relevant to proxy scenario?

All that said - I think most likely it's best to keep it simple like it is here.

Reply:
Currently refactoring it:
To 1: I've refactored it to have the properties below the proxy property
To 2: There is a .Default method in connection wich will be called recursively when using the Connection as a type for the proxy property as such I've created a new type called ProxyConnection (the Connection type has got the proxy field)
To 3: Yes they're relevant, currently in progress to implement them